staging/src/k8s.io/api/batch/v1/types.go

```diff
@@ -204,6 +207,7 @@ type PodFailurePolicyOnExitCodesRequirement struct {
 type PodFailurePolicyOnPodConditionsPattern struct {
 	// Specifies the required Pod condition type. To match a pod condition
 	// it is required that specified type equals the pod condition type.
+	// +required
```

Comment:
Validates as a qualified names, which must not be empty
https://github.com/kubernetes/kubernetes/blob/e34f01402f2be14e68b03a58c054f1a86bff55b9/pkg/apis/batch/validation/validation.go#L364
Agreed required